pycbc/inference/models/tools.py

```diff
@@ -378,7 +405,9 @@ def marginalize_likelihood(sh, hh,
             else:
                 vweights = dist_weights
 
-        if phase:
+        if return_complex:
+            return sh, -0.5 * hh
+        elif phase:
```

Comment:
A little off topic, could you add a comment here explaining why you need to add `sh` to the `log(i0e(sh))`? Julian was asking me the other day, and it took me awhile to recall (it's because we want log(I0(sh)), but the scipy `i0e` function returns `exp(-|sh|)I0(sh))`.

Actually, now that I think of it... there might be a bug here. I think that should be `+ abs(sh)`. According to the [scipy docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.i0e.html), `i0e` returns `exp(-abs(x))*i0(x)`. That means if `sh` is negative, then we'll end up returning `log(i0e(sh)) - 2sh`. I'm not sure why this hasn't bitten us before; maybe because sh is positive for things we care about? Still, that would affect evidence calculations.

Reply:
@cdcapano If you calculate the marginalization, you have to take |sh| as the argument, so it is never negative. Look earlier in the code, this is already done at line  https://github.com/gwastro/pycbc/blob/c809928a2cec48afdbed7a841990462ea8926eae/pycbc/inference/models/tools.py#L382. I guess 412/413 is redundant though, which may have caused the confusion here.

Reply:
Oh, I missed that. Ok, it's fine then. Man, the logic in this function is complicated.